### PR TITLE
fix: preserve unknown fields value

### LINKED ISSUE #1338: Frontmatter should retain unknown fields
Currently, frontmatter fields not defined in the `fields` option are removed when editing a file.
I don't know the rationale behind this behaviour, but to me (and others #1174) it is highly unexpected. This can easily break things when intended frontmatter configuration gets silently stripped off with no immediate benefit.
In my opinion it would make more sense to simply ignore and keep frontmatter fields the CMS doesn't know about.

I often use frontmatter fields to define specific configuration of a page that is not supposed to be editable through the CMS. While it is possible to add `hidden` fields for that, it seems highly unnecessary - and is very tedious if you have to repeat multiple hidden fields for each file in a file collection. And it's still error prone because you easily forget to add a custom field name to the config. When somebody edits that file through the CMS, it's broken.

### PR DESCRIPTION
**Pull request by barthc**
Closes #1338

### CODE REVIEW

File: packages/netlify-cms-core/src/actions/editorialWorkflow.ts [hunk 1]
```diff
@@ -386,7 +386,9 @@ export function persistUnpublishedEntry(collection: Collection, existingUnpublis
      * update the entry and entryDraft with the serialized values.
      */
     const fields = selectFields(collection, entry.get('slug'));
-    const serializedData = serializeValues(entry.get('data'), fields);
+    let serializedData = serializeValues(entry.get('data'), fields);
```
Comment (erezrokah) [inline packages/netlify-cms-core/src/actions/editorialWorkflow.ts]: We should apply this fix also here:
https://github.com/netlify/netlify-cms/blob/684b79e43bebb63ce1e844eae5c8c0e76087687b/packages/netlify-cms-core/src/actions/entries.ts#L617

Right?

Also, what do you think about moving this fix into the `serializeValues` and `deserializeValues` functions? 

Should be easier to add unit tests for that as well.

File: packages/netlify-cms-core/src/formats/helpers.js [hunk 2]
```diff
@@ -1,12 +1,7 @@
 export const sortKeys = (sortedKeys = []) => (a, b) => {
   const idxA = sortedKeys.indexOf(a);
   const idxB = sortedKeys.indexOf(b);
-  if (idxA === -1 || idxB === -1) {
```
Comment (erezrokah) [inline packages/netlify-cms-core/src/formats/helpers.js]: Any reason to remove this?

```js
yaml.toFile({ title: 'title', unknown1: 'unknown1', body: 'body', unknown2: 'unknown2' }, [
          'title',
          'body',
        ])

yaml.toFile({ title: 'title', body: 'body', unknown1: 'unknown1', unknown2: 'unknown2' }, [
          'title',
          'body',
        ])
```
Will generate different results.

**Comment by erezrokah** [comment]
Thanks for this @barthc, very important issue to fix.

File: packages/netlify-cms-core/src/formats/helpers.js (hunk 2 above)
Comment (barthc) [reply packages/netlify-cms-core/src/formats/helpers.js]: In fact, I think we should remove the sort and preserve the original field order.  If there is an unknown field, the field order is rearranged. I will be happy to leave it as is if users don't mind.

Comment (erezrokah) [reply packages/netlify-cms-core/src/formats/helpers.js]: I was actually thinking the same thing about the sorting as I'm not sure why its there in the first place (maybe @erquhart can help), but as long as we have it, it should be consistent.

**Comment by erezrokah** [comment]
I created new branch with an additional commit https://github.com/netlify/netlify-cms/commit/811d4844d4168e45eef5005f42b950e289d8df31 per my comments.
If you're good with it I can push it to this branch and merge the PR.

File: packages/netlify-cms-core/src/formats/helpers.js (hunk 2 above)
Comment (barthc) [reply packages/netlify-cms-core/src/formats/helpers.js]: This is what the sort is for, on new entry if an optional field is left untouched, the value is not add on the frontmatter, on subsequent edits when the field value is entered this sort helps to move it to the right position.

Comment (barthc) [reply packages/netlify-cms-core/src/formats/helpers.js]: It would make sense to do the same for the unknown fields.

Comment (erezrokah) [reply packages/netlify-cms-core/src/formats/helpers.js]: Oh, I get it now.

Comment (erezrokah) [reply packages/netlify-cms-core/src/formats/helpers.js]: Ok I think this is good to go with https://github.com/netlify/netlify-cms/commit/4e23668412e6d05ccd61954a9abd9ba3d5e395a7

What do you think?

Comment (barthc) [reply packages/netlify-cms-core/src/formats/helpers.js]: Yes, I think so too.

**Comment by erezrokah** [comment]
GitHub action got stuck on retrieving yarn cache. Retrying

**Pull request merged**